src/BaseConnector.ts

```diff
@@ -18,46 +20,46 @@ class BaseConnector<ConfigOptionsType = any, EventOptionsType = any> {
     app.register(this)
   }
 
-  updateOptions(configOptions: ConfigOptionsType) {
+  updateOptions(configOptions: ConfigOptionsType): void {
     // Override this method if you need to do something specific on update configOptions
     this.configOptions = configOptions
   }
 
-  removeEvent(event: EventConfiguration) {
+  removeEvent(event: EventConfiguration): void {
     delete this.eventConfigurations[event.id]
     this.onRemoveEvent(event)
   }
 
-  onRemoveEvent(event: EventConfiguration) {
+  onRemoveEvent(event: EventConfiguration): void {
     // Override this method if you need to do something specific on remove events
   }
 
-  start() {
+  start(): void {
     if (!this.started) {
       this.onStart()
     }
     this.started = true
   }
 
-  onStart() {
+  onStart(): void {
     // Override this method if you need to do something specific on start
   }
 
   on(
     options: EventOptionsType,
-    handler?: any,
+    handler?: Handler | HandlerWrapper,
     eventId?: EventConfiguration['id'],
   ): EventConfiguration | null {
     console.log('The on method is not implemented for this connector')
     return null
   }
 
-  stop() {
+  stop(): Promise<void> | void {
```

Comment:
I think stop needs to be an async function and onStop too. Then the output type would be Promise<void> only

Reply:
same for start/onStart

Reply:
Some connectors wont have async onStop functions though: https://github.com/reshufflehq/reshuffle/blob/1d129ceea151066490c15a32fc967d44cd799dbe/src/connectors/CronConnector.ts#L56